File: Entra App Secret/entra_app_secret.py
# Using MSAL for Authentication:
import msal
import requests

client_id = "your_client_id"
client_secret = "your_client_secret"
tenant_id = "your_tenant_id"
authority = f"https://login.microsoftonline.com/{tenant_id}"
scope = ["https://graph.microsoft.com/.default"]

app = msal.ConfidentialClientApplication(
    client_id, authority=authority, client_credential=client_secret
)

token_response = app.acquire_token_for_client(scopes=scope)
access_token = token_response.get("access_token")

headers = {"Authorization": "Bearer " + access_token}
response = requests.get("https://graph.microsoft.com/v1.0/me", headers=headers)

print(response.json())


# Using Azure Identity for Azure SDKs:
from azure.identity import ClientSecretCredential
from azure.keyvault.secrets import SecretClient

tenant_id = "your_tenant_id"
client_id = "your_client_id"
client_secret = "your_client_secret"
vault_url = "https://your-key-vault-name.vault.azure.net/"

credential = ClientSecretCredential(tenant_id, client_id, client_secret)
client = SecretClient(vault_url=vault_url, credential=credential)

secret = client.get_secret("your-secret-name")
print(secret.value)


# Using Azure Management Libraries:
from azure.identity import ClientSecretCredential
from azure.mgmt.resource import ResourceManagementClient

tenant_id = "your_tenant_id"
client_id = "your_client_id"
client_secret = "your_client_secret"
subscription_id = "your_subscription_id"

credential = ClientSecretCredential(tenant_id, client_id, client_secret)
resource_client = ResourceManagementClient(credential, subscription_id)

for resource_group in resource_client.resource_groups.list():
    print(resource_group.name)

File: Entra App Secret/entra_app_secret.js
// Using @azure/identity with @azure/keyvault-secrets:
const { ClientSecretCredential } = require("@azure/identity");
const { SecretClient } = require("@azure/keyvault-secrets");

const tenantId = "your_tenant_id";
const clientId = "your_client_id";
const clientSecret = "your_client_secret";
const vaultUrl = "https://your-key-vault-name.vault.azure.net/";

const credential = new ClientSecretCredential(tenantId, clientId, clientSecret);
const client = new SecretClient(vaultUrl, credential);

async function getSecret() {
    const secret = await client.getSecret("your-secret-name");
    console.log(secret.value);
}

getSecret();




// Using @azure/msal-node for Authentication:
const { ConfidentialClientApplication } = require("@azure/msal-node");

const config = {
    auth: {
        clientId: "your_client_id",
        authority: "https://login.microsoftonline.com/your_tenant_id",
        clientSecret: "your_client_secret",
    },
};

const client = new ConfidentialClientApplication(config);

const tokenRequest = {
    scopes: ["https://graph.microsoft.com/.default"],
};

client.acquireTokenByClientCredential(tokenRequest).then((response) => {
    console.log(response.accessToken);
}).catch((error) => {
    console.error(error);
});


// Using axios to Access Azure Resources:
const axios = require('axios');
const { ConfidentialClientApplication } = require("@azure/msal-node");

const config = {
    auth: {
        clientId: "your_client_id",
        authority: "https://login.microsoftonline.com/your_tenant_id",
        clientSecret: "your_client_secret",
    },
};

const client = new ConfidentialClientApplication(config);

const tokenRequest = {
    scopes: ["https://management.azure.com/.default"],
};

async function getResourceGroups() {
    try {
        const response = await client.acquireTokenByClientCredential(tokenRequest);
        const token = response.accessToken;

        const resourceResponse = await axios.get("https://management.azure.com/subscriptions/your_subscription_id/resourcegroups?api-version=2021-04-01", {
            headers: {
                Authorization: `Bearer ${token}`,
            },
        });

        console.log(resourceResponse.data);
    } catch (error) {
        console.error(error);
    }
}

getResourceGroups();